packages/manager/src/features/CloudPulse/Alerts/CreateAlert/Criteria/DimensionFilterField.tsx

```diff
@@ -0,0 +1,177 @@
+import DeleteOutlineOutlined from '@mui/icons-material/DeleteOutlineOutlined';
+import { Grid, styled } from '@mui/material';
+import React from 'react';
+import { Controller, useFormContext } from 'react-hook-form';
+
+import { Autocomplete } from 'src/components/Autocomplete/Autocomplete';
+
+import { DimensionOperatorOptions } from '../../constants';
+
+import type { Dimension } from '@linode/api-v4';
+
+interface DimensionFilterFieldProps {
+  /**
+   * dimension filter data options to list in the Autocomplete component
+   */
+  dimensionOptions: Dimension[];
+  /**
+   * name (with the index) used for the component to set in form
+   */
+  name: string;
+  /**
+   * function to delete the DimensionFilter component
+   * @returns void
+   */
+  onFilterDelete: () => void;
+}
+type DimensionDataFieldOption = {
+  label: string;
+  value: string;
+};
+export const DimensionFilterField = (props: DimensionFilterFieldProps) => {
+  const { dimensionOptions, name, onFilterDelete } = props;
+
+  const { control, setValue, watch } = useFormContext();
+
+  const dataFieldOptions = dimensionOptions.map((dimension) => ({
+    label: dimension.label,
+    value: dimension.dimension_label,
+  }));
+
+  const [
+    selectedDataField,
+    setSelectedDataField,
+  ] = React.useState<DimensionDataFieldOption | null>(null);
+
+  const handleSelectChange = (
+    field: string,
+    value: string,
+    operation: string
+  ) => {
+    if (operation === 'selectOption') {
+      setValue(`${name}.${field}`, value);
+    } else {
+      setValue(`${name}.${field}`, '');
```

Comment:
we can put condition for clear option for reseting the selected value, rather than putting it in else. can discuss over this.